### PR TITLE
Extend public API and fix documentation

### PR DESCRIPTION
**Pull request by ericpre**
Add back some functions which were removed when splitting from hyperspy. Some modules and functions that were not to the public API as listed in the [API reference](https://hyperspy.org/exspy/reference/index.html) have been removed.

### Progress of the PR
- [x] Add some functionalities back to public API,
- [x] Rename private modules to start with underscore to reduce their visibility and make more clear that they are not part of the public API, 
- [x] docstring updated: there were still a lot of leftover from hyperspy,
- [x] update user guide: there were still a lot of leftover from hyperspy,
- [n/a] added tests,
- [x] add a changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/exspy/blob/main/upcoming_changes/README.rst)),
- [x] Check formatting of the changelog entry (and eventual user guide changes) in the `docs/readthedocs.org:exspy` build of this PR (link in github checks)
- [x] ready for review.

### CODE REVIEW

**Comment by CSSFrancis** [comment]
@ericpre I can review this today!

File: doc/user_guide/eds.rst [hunk 1]
```diff
@@ -59,7 +60,7 @@ dimensional signal with the energy axis in first position):
     >>> si
     <Signal1D, title: , dimensions: (256, 224|1024)>
 
-Finally, for a stack of spectrum images, using "*" as a wildcard character:
+Finally, for a stack of spectrum images, using ``"*"`` as a wildcard character:
```
Comment (CSSFrancis) [inline doc/user_guide/eds.rst:63]:
```suggestion
Finally, for a stack of spectrum images, ``"*"`` can be used as a wildcard character:
```

**Review by CSSFrancis — APPROVED** [review]
These changes look good to me.  And probably should have been done a little bit ago. 

My one hesitation is that this will move all of the functions in the `misc` to a different place.  Tecnically I think these were part of the public api even if they weren't documentated.  It might be worthwhile to create a `misc.py` file which redirects to the `_misc.py` file and then raises a deprecation warning. 

That's what we have been doing in pyxem just to catch people.

**Comment by ericpre** [comment]
Thank you @CSSFrancis for the review. I added a shim for the functionalities that have been added to the public API. Do you want to have another look?

**Review by CSSFrancis — APPROVED** [review]
@ericpre Looks great!

**Pull request merged**